agent/engine/docker_image_manager.go

```diff
@@ -72,6 +74,18 @@ func NewImageManager(cfg *config.Config, client DockerClient, state dockerstate.
 	}
 }
 
+// PreserveImage add the image into whitelists that won't be cleaned up by agent
```

Comment:
PreserveImage adds the image into the preservedImages array...

Separately, is it worth including the disclaimer about why it is unsafe in the doc string or in a separate comment?